### PR TITLE
cherry pick CLIWrapper deadlock fix

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

cherry pick fixes from https://github.com/DynamoDS/Dynamo/pull/14649/ for CLIWrapper classes

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoUtilities/Md2Html.cs [hunk 1]
```diff
@@ -104,7 +102,7 @@ internal string SanitizeHtml(string content)
 
             var output = GetData(processCommunicationTimeoutms);
 
-           return output.Result;
+           return output;
```
Comment (aparajit-pratap) [inline src/DynamoUtilities/Md2Html.cs:105]: how about `return GetData(processCommunicationTimeoutms);`

**Review by aparajit-pratap — APPROVED** [review]

File: src/DynamoUtilities/CLIWrapper.cs [hunk 2]
```diff
@@ -145,7 +145,7 @@ protected virtual async Task<string> GetData(int timeoutms)
                     return writer.ToString();
                 }
             });
-            var completedTask = await Task.WhenAny(readStdOutTask, Task.Delay(TimeSpan.FromMilliseconds(timeoutms)));
+            var completedTask = Task.WhenAny(readStdOutTask, Task.Delay(TimeSpan.FromMilliseconds(timeoutms))).Result;
             //if the completed task was our read std out task, then return the data
             //else we timed out, so return an empty string.
             return completedTask == readStdOutTask ? readStdOutTask.Result : string.Empty;
```
Comment (pinzart90) [inline src/DynamoUtilities/CLIWrapper.cs:151]: we might be able to simplify lines 148 to 151 by using
```
return readStdOutTask.Wait(timeoutms) ? readStdOutTask.Result : string.Empty;
```
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.wait?view=net-8.0#system-threading-tasks-task-wait(system-int32)

Comment (pinzart90) [reply src/DynamoUtilities/CLIWrapper.cs:151]: either way we go, please test that the timeout can be reached without issues.
Ex Add a Thread.Sleep(timeoutms *2) in the TaskRun delegate.

Comment (mjkkirschner) [reply src/DynamoUtilities/CLIWrapper.cs:151]: @pinzart90 - does this test already cover what you are concerned about -or not?
https://github.com/DynamoDS/Dynamo/blob/master/test/Libraries/DynamoUtilitiesTests/CLIWrapperTests.cs#L47

Comment (mjkkirschner) [reply src/DynamoUtilities/CLIWrapper.cs:162]: @pinzart90 I've added another test and slightly modified the GetData method to make it possible to inject a mock readLine delegate that will be invoked inside the Task delegate so we can sleep or return data etc.

**Review by sm6srw — APPROVED** [review]

**Review by pinzart90 — APPROVED** [review]

**Comment by mjkkirschner** [comment]
I am going to merge this.
1. it fixes the deadlock on master
2. The failing test is on master already - unclear how it did not fail earlier test runs.
3. I will try to add a regression test for this deadlock in another PR.

**Pull request merged**